src/containers/ListingPage/ListingPage.duck.js

```diff
@@ -406,15 +410,19 @@ export const loadData = (params, search, config) => (dispatch, getState, sdk) =>
     return Promise.resolve();
   }
 
-  return Promise.all([
-    dispatch(showListing(listingId, config)),
-    dispatch(fetchReviews(listingId)),
-  ]).then(response => {
+  const hasViewingRights = currentUser && hasPermissionToViewData(currentUser);
+  const promises = hasViewingRights
+    ? // For users with viewing rights, fetch the listing and the associated reviews
+      [dispatch(showListing(listingId, config)), dispatch(fetchReviews(listingId))]
+    : // If user has no viewing rights, only allow fetching their own listing without reviews
+      [dispatch(showListing(listingId, config, true))];
```

Comment:
Sorry, one more thing.
This can't default to ownListings endpoint on server because then the server can't return correct HTTP headers on public marketplace scenario. I.e. if listing is not found from public endpoint listings.show, the 404 should be set.

Reply:
Fixed ListingPage.duck.js condition to default to listings.show, and updated the listing pages condition accordingly